cli/src/org/partiql/cli/prettyprintAST/FeatureTree.kt

```diff
@@ -0,0 +1,51 @@
+package org.partiql.cli.prettyprintAST
+
+import org.partiql.lang.domains.PartiqlAst
+
+/**
+ * PIG AST is not a recursive data structure, thus it is not easy to transform it directly to a pretty printed string.
+ * So we need to first transform it into FeatureTree, which is a recursive tree structure (it has a list of children which are also Feature Tree),
+ * then we can recursively pretty print the FeatureTree as we want.
+ *
+ * @param astType is a string of the PIG AST node type
+ * @param value is the value in case node type is [PartiqlAst.Expr.Lit]
+ * @param attrOfParent is a string which represents which attribute it belongs to its parent
+ * @param children is a list of child FeatureTree
+ *
+ * The first 3 parameters are chosen as they represent the main feature of a PIG AST node.
+ *
+ */
+class FeatureTree(astType: String, value: String? = null, attrOfParent: String? = null, children: List<FeatureTree>? = null) {
```

Comment:
Why the name "feature"?  Doesn't really seem related to what this actually doing.

Why is `FeatureTree` better than s-expressions?

In general, there is going to be a long term maintenance cost to this, since it touches every node in the AST.  Is there a way to avoid having to touch every single node?

Reply:
This is my COW task. 

1. What's your suggested name for this tree structure? (I'm not good at naming : ) )
2. As described in https://github.com/partiql/partiql-lang-kotlin/issues/546, I think the current s-expressions for a parsed tree is not readable enough. 
3. Since the PIG AST does not have something like a list of children, I think we have to touch each node, otherwise, there is no way to know the child tree nodes for each AST node.